### PR TITLE
[WIP] Sign beta versions (bug 1160593)

### PR DESCRIPTION
**Pull request by robhudson**
I'm attempting to get this to:

* Sign new beta files with the prelim cert
* Add them to the prelim queue

I'm getting hung up on the tests. I'll keep trying to work through it but something somewhere is changing statuses on me and I can't find it. Maybe a signal somewhere that isn't clear.

### CODE REVIEW

File: apps/devhub/forms.py [hunk 1]
```diff
@@ -537,8 +537,9 @@ class NewVersionForm(NewAddonForm):
         })
     beta = forms.BooleanField(
         required=False,
-        help_text=_lazy(u'A file with a version ending with a|alpha|b|beta and'
-                        u' an optional number is detected as beta.'))
+        help_text=_lazy(u'A file with a version ending with '
+                        u'a|alpha|b|beta|pre|rc and an optional number is '
+                        u'detected as beta.'))
```
Comment (magopian) [inline apps/devhub/forms.py]: :+1:

File: apps/devhub/views.py [hunk 2]
```diff
@@ -1233,25 +1232,27 @@ def version_add(request, addon_id, addon):
...
     storage.delete(upload.path)
-    check_validation_override(request, form, addon, new_file.version)
+    check_validation_override(request, new_file_form, addon, new_file.version)
     file_form = forms.FileFormSet(prefix='files', queryset=version.files.all())
     form = [f for f in file_form.forms if f.instance == new_file]
 
     # Not listed? Do we need to sign?
     if (waffle.flag_is_active(request, 'automatic-validation') and
             not new_file.version.addon.is_listed):
+        print "A"
```
Comment (magopian) [inline apps/devhub/views.py]: Debug statement ;)

File: apps/devhub/views.py [hunk 3]
```diff
@@ -1259,6 +1260,9 @@ def version_add_file(request, addon_id, addon, version_id):
             # Passed validation: sign automatically without manual review.
             new_file.update(status=amo.STATUS_LITE)
             sign_file(new_file)
+    elif new_file_form.cleaned_data.get('beta'):
```
Comment (magopian) [inline apps/devhub/views.py]: At the moment, nothing prevents a fully reviewed unlisted addon from having beta versions, so we need to either:
1/ prevent it (not displaying the "this is a beta" checkbox in the submission form, not detecting it as beta)
2/ allow it, but then also deal with the case where the beta version is for an unlisted addon

File: apps/devhub/views.py [hunk 4]
```diff
@@ -1259,6 +1260,9 @@ def version_add_file(request, addon_id, addon, version_id):
...
+        # Beta files always get signed with prelim cert.
+        sign_file(new_file)
```
Comment (magopian) [inline apps/devhub/views.py]: That's not what we want: we don't want to sign them automatically, unless they've passed automatic validation. If they don't, they should only be queue for prelim review.

We don't have a status for "beta version waiting for review", so I think we're going to need a new one. I'll post a comment in the bug.

File: lib/crypto/packaged.py [hunk 5]
```diff
@@ -86,8 +87,9 @@ def sign_file(file_obj):
         log.info('Not signing file {0}: addon is a hotfix'.format(file_obj.pk))
         return
 
-    # We only sign files that have been reviewed.
-    if file_obj.status not in amo.REVIEWED_STATUSES:
+    # We only sign files that have been reviewed plus beta files.
+    # TODO: Does this make sense? Check repurcussions of this.
+    if file_obj.status not in amo.REVIEWED_STATUSES + (amo.STATUS_BETA,):
```
Comment (magopian) [inline lib/crypto/packaged.py]: We might want another status for beta files that aren't reviewed yet. We could then simply add `amo.STATUS_BETA` to the list of `amo.REVIEWED_STATUSES`.

**Comment by robhudson** [comment]
The bug requirements changed quite a bit so closing this.

**Pull request closed without merging**